spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/LocalAppDeployer.java

```diff
@@ -229,12 +225,7 @@ public DeploymentState getState() {
 			Integer exit = getProcessExitValue(process);
 			// TODO: consider using exit code mapper concept from batch
 			if (exit != null) {
-				if (exit == 0) {
```

Comment:
My concern was that the deployer should be left in the exact same state after a successful undeployment as it was before the app was ever known (hence state == `unknown`). After careful reading of the code, this is ok, as the key is removed from the map after undeployment.
So, greenlight for me. (even tested with SCDF)